lib/http.c

```diff
@@ -3387,11 +3387,12 @@ CURLcode Curl_http_readwrite_headers(struct Curl_easy *data,
         }
       }
       else if(conn->handler->protocol & CURLPROTO_RTSP) {
+        int wall;
         nc = sscanf(HEADER1,
-                    " RTSP/%d.%d %3d",
+                    " RTSP/%1d.%1d %3d %d",
                     &rtspversion_major,
                     &conn->rtspversion,
-                    &k->httpcode);
+                    &k->httpcode, &wall);
         if(nc == 3) {
```

Comment:
Is this still going to match? If all fields match, isn't sscanf going to return 4 instead?

Reply:
sscanf() is a bit tricky to use for this matching. It doesn't really care about the spaces in the match string so an input like `1.1234567 200` (ie an illegal string) will match `%1d.%d %3d` and store 1, 1 and 234 and return 3. I don't think we want that.

So with `%1d.%d %3d %d` I want the above input to store 1,1,234, 567 and return 4, which isn't and shouldn't be a match.

While explaining this I think the case where the input says `1.1 200 200 reasons to be fine` is also going to not match while it is a fine input. The second "200" just happens to be part of the string...

I'll improve the logic.